core/src/main/scala/org/http4s/rho/bits/EntityParser.scala

```diff
@@ -0,0 +1,49 @@
+package org.http4s
+package rho.bits
+
+import scala.language.higherKinds
+
+import scala.annotation.tailrec
+import scala.collection.generic.CanBuildFrom
+import org.http4s.EntityDecoder
+import org.http4s.DecodeFailure
+import scalaz.concurrent.Task
+
+/** Extract a value from the `Request` `Query`
+  *
+  * @tparam A Type of value produced by the parser.
+  */
+ trait EntityParser[A] { self => 
```

Comment:
I don't know this to be true, but I suspect there is a way to do this without adding a wrapper around `EntityDecoder`. I haven't actually explored that, but if it does turn out to be the case, I'd like to avoid a wrapper type. It might be a simple as treating `EntityDecoder[SomeExceptionWrapper\/T]` special and then an `EntityDecoder[T]` could be lifted up inside another `EntityDecoder`.